### ISSUE TITLE
Separating positive and negative test cases for rules

### ISSUE THREAD

**Issue by schalkms** [arturbosch/detekt#527]
At the moment the tests check the violation count.
However, there can be a case where the sum is correct and the findings are wrong.
Separating positive and negative test cases could help to minimize this problem.

**Comment by arturbosch** [comment]
Which test cases do you have in mind? I'm not a fan of refactoring just because of refactoring. If the logic gets changed, we can change some test cases?

**Comment by schalkms** [comment]
Consider the following test file example for checking the naming convention.
```kotlin
class MyClass {}//valid
class myclass {} // report
class myClass {} // report
```
Detekt's current test case checks the violation count. However, detekt does not check if the wrong **MyClass** classes are reported. There could be case where the right one gets reported.

**Comment by arturbosch** [comment]
Ah now I understand. Yes, that makes sense and should be the preferred way.

**Comment by schalkms** [comment]
Closed with 0d869e3
